cmd/statusd/server/parse-api/main.go

```diff
@@ -0,0 +1,158 @@
+		return
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	var publicFunctionsWithArgs []string
+	var publicFunctionsWithoutArgs []string
+	var unsupportedFunctions []string
+	var deprecatedFucntions []string
+	var isDeprecated bool
+	var packageName string
+
+	for scanner.Scan() {
+		line := strings.TrimSpace(scanner.Text())
+
+		// Detect package name
+		if strings.HasPrefix(line, "package ") {
+			packageName = strings.TrimPrefix(line, "package ")
+		}
+
+		// Check for deprecation comment
+		if isDeprecatedComment(line) {
+			isDeprecated = true
+			continue
+		}
+
+		if !publicFunc.MatchString(line) {
+			continue
+		}
+
+		functionName := extractFunctionName(line)
+
+		if isDeprecated {
+			isDeprecated = false
+			deprecatedFucntions = append(deprecatedFucntions, functionName)
+		}
+
+		switch {
+		case isPublicFunctionWithArgs(line):
+			publicFunctionsWithArgs = append(publicFunctionsWithArgs, functionName)
+		case isPublicFunctionWithoutArgs(line):
+			publicFunctionsWithoutArgs = append(publicFunctionsWithoutArgs, functionName)
+		default:
+			unsupportedFunctions = append(unsupportedFunctions, functionName)
+		}
+	}
+
+	if err := scanner.Err(); err != nil {
+		fmt.Printf("Error reading file: %s\n", err)
+		return
+	}
+
+	// Prepare the template data
+	data := TemplateData{
+		PackageName:          packageName,
+		FunctionsWithResp:    publicFunctionsWithArgs,
+		FunctionsNoArgs:      publicFunctionsWithoutArgs,
+		UnsupportedEndpoints: unsupportedFunctions,
+		DeprecatedEndpoints:  deprecatedFucntions,
+	}
+
+	// Load and parse the template
+	tmpl, err := template.ParseFiles(templateFilePath)
+	if err != nil {
+		fmt.Printf("Failed to parse template file: %s\n", err)
+		return
+	}
+
+	// Create the output file
+	outputFile, err := os.Create(outputFilePath)
+	if err != nil {
+		fmt.Printf("Failed to create output file: %s\n", err)
+		return
+	}
+	defer outputFile.Close()
+
+	// Execute the template and write the result to the output file
+	err = tmpl.Execute(outputFile, data)
+	if err != nil {
+		fmt.Printf("Failed to execute template: %s\n", err)
+		return
+	}
+
+	fmt.Println("Generated endpoints file:", outputFilePath)
+}
+
+// Function to check if a line contains a public function with a response of string
+func isPublicFunctionWithArgs(line string) bool {
+	return publicFuncWithArgsPattern.MatchString(line)
+}
+
+// Function to check if a line contains a public function with not arguments and a response of string
+func isPublicFunctionWithoutArgs(line string) bool {
+	return publicFuncWithoutArgsPattern.MatchString(line)
+}
+
+// Function to extract the public function name from a line
+func extractFunctionName(line string) string {
+	matches := funcNamePattern.FindStringSubmatch(line)
+	if len(matches) > 1 {
+		return matches[1]
+	}
+	return ""
+}
+
+// Function to check if a comment indicates a deprecated function
+func isDeprecatedComment(line string) bool {
+	return strings.Contains(line, "// Deprecated:")
```

Comment:
sometimes it's `// DEPRECATED`

Reply:
Done: 2ba26131120b1a6c2d5ae3fd783e54e1c23fa797

Checking it like this:
https://github.com/status-im/status-go/blob/2ba26131120b1a6c2d5ae3fd783e54e1c23fa797/cmd/statusd/server/parse-api/main.go#L35

Which matches https://go.dev/wiki/Deprecated.
Not bothering about `@deprecated` form, as it's not conventional in Go. Anyway, it's only used in `wallet/api.go`, and is not parsed by this program.